src/sinks/util/http.rs

```diff
@@ -175,7 +177,12 @@ where
 
     fn start_send(self: Pin<&mut Self>, mut event: Event) -> Result<(), Self::Error> {
         let finalizers = event.metadata_mut().take_finalizers();
+        let byte_size = event.size_of();
```

Comment:
Is this right? I think for sinks we will want the serialized bytes rather than the in-memory byte count. From the [component spec](https://github.com/vectordotdev/vector/blob/master/docs/specs/component.md#bytessent):

> For HTTP-based protocols, the total number of bytes in the HTTP body, as represented by the Content-Length header.

Reply:
This is used by the `EventsSent` event, which is speced to use the in-memory byte size.

Reply:
Ah, yes, I see 😄